### DISCUSSION TITLE
Kserve Predict Protocol V2 Requirement ( /infer )

### DISCUSSION THREAD

**Discussion by oyy2000** [kubeflow/kfserving#2369] [Q&A]
Hello, I need to extend our framework of model to support the Kserve Predict Protocol - Version 2. When implementing the infer API, I came up with some questions:
1. Why will there be two input but only one output in a certain model.
2. How will the two inputs in a difference datatype make sense.

![image](https://user-images.githubusercontent.com/56588848/182593550-5cc50e68-ab68-4dd1-9471-11f7a0140ac3.png)
The figure is from https://github.com/kserve/kserve/blob/master/docs/predict-api/v2/required_api.md#tensor-data

**Comment by yuzisun** [comment]
@oyy2000 both `inputs` and `outputs` are array, what shows above is an example. 

see https://github.com/kserve/kserve/blob/master/docs/predict-api/v2/required_api.md#inference-request-json-object

**Reply by oyy2000** [reply]
Yes, I wonder if we should response each input with a correspond output?

**Reply by yuzisun** [reply]
Not necessarily, the number of outputs can be totally different from the number of inputs depending on your model .
